src/devices/device_models/thermal_generation.jl

```diff
@@ -184,44 +203,95 @@ end
 
 # ramping constraints
 
-"""
-This function adds the ramping limits of generators when there are CommitmentVariables
-"""
-function ramp_constraints(ps_m::CanonicalModel, devices::Array{T,1}, device_formulation::Type{D}, system_formulation::Type{S}, time_range::UnitRange{Int64}, initial_conditions::Array{Float64,1}) where {T <: PSY.ThermalGen, D <: AbstractThermalFormulation, S <: PM.AbstractPowerFormulation}
+function _get_data_for_rocc(devices::Array{T,1}) where {T <: PSY.ThermalGen}
+        
+    set_name = Vector{String}(undef, length(devices))
+    ramp_params = Vector{UpDown}(undef, length(devices))     
+    minmax_params = Vector{MinMax}(undef, length(devices))                                                              
+
+    idx = eachindex(devices)
+    i, state = iterate(idx)
+    for g in devices
+        if !isnothing(g.tech.ramplimits)
+            set_name[i] = g.name
+            ramp_params[i] = g.tech.ramplimits
+            minmax_params[i] = g.tech.activepowerlimits
+            i, state = iterate(idx, state)
+            state === nothing && (i += 1; break)                                                                   
+        end 
+    end
+    
+    deleteat!(set_name, i:last(idx))
+    deleteat!(ramp_params, i:last(idx))
+    deleteat!(minmax_params, i:last(idx))
```

Comment:
These make the T+1 time period nonbinding?

Reply:
Line 220 adds a +1 to i to avoid that